File: libraries/leoz-common/src/main/java/org/deku/leoz/messaging/log/LogListener.java
package org.deku.leoz.messaging.log;

import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.deku.leoz.messaging.MessagingContext;
import org.deku.leoz.messaging.log.v1.LogMessage;
import sx.jms.Handler;
import sx.jms.listeners.SpringJmsListener;
import sx.jms.converters.DefaultConverter;

import javax.jms.Destination;
import javax.jms.JMSException;
import javax.jms.Message;
import javax.jms.Session;
import java.time.Instant;
import java.time.LocalDateTime;
import java.time.ZoneId;

/**
 * Leo2 log message listener
 * Created by masc on 16.04.15.
 */
public class LogListener extends SpringJmsListener implements Handler<LogMessage[]> {
    private Log mLog = LogFactory.getLog(this.getClass());
    private MessagingContext mMessagingContext;

    public LogListener(MessagingContext messagingContext) {
        super(messagingContext.getBroker().getConnectionFactory());
        mMessagingContext = messagingContext;

        this.setConverter(new DefaultConverter(
                DefaultConverter.SerializationType.KRYO,
                DefaultConverter.CompressionType.GZIP));

        this.addDelegate(LogMessage[].class, this);
    }

    @Override
    protected Destination createDestination() {
        return mMessagingContext.getCentralLogQueue();
    }

    @Override
    public void onMessage(LogMessage[] message, Message jmsMessage, Session session) throws JMSException {
        mLog.info(String.format("message id [%s] %s",
                jmsMessage.getJMSMessageID(),
                LocalDateTime.ofInstant(
                        Instant.ofEpochMilli(
                                jmsMessage.getJMSTimestamp()), ZoneId.systemDefault())));
    }
}

File: libs/leoz-common/src/main/java/org/deku/leoz/messaging/log/LogListener.kt
package org.deku.leoz.messaging.log

import org.apache.commons.logging.Log
import org.apache.commons.logging.LogFactory
import org.deku.leoz.messaging.MessagingContext
import sx.jms.Handler
import sx.jms.listeners.SpringJmsListener
import sx.jms.converters.DefaultConverter

import javax.jms.Destination
import javax.jms.JMSException
import javax.jms.Message
import javax.jms.Session
import java.time.Instant
import java.time.LocalDateTime
import java.time.ZoneId

/**
 * Log message listener
 * Created by masc on 16.04.15.
 */
class LogListener(
        /** Messaging context */
        private val messagingContext: MessagingContext) : SpringJmsListener(messagingContext.broker.connectionFactory), Handler<Array<LogMessage>> {

    init {
        this.converter = DefaultConverter(
                DefaultConverter.SerializationType.KRYO,
                DefaultConverter.CompressionType.GZIP)

        this.addDelegate(Array<LogMessage>::class.java, this)
    }

    override fun createDestination(): Destination {
        return messagingContext.centralLogQueue
    }

    override fun onMessage(message: Array<LogMessage>, jmsMessage: Message, session: Session) {
        val timestamp = LocalDateTime.ofInstant(
                Instant.ofEpochMilli(
                        jmsMessage.jmsTimestamp), ZoneId.systemDefault())

        log.info("message id [${jmsMessage.jmsMessageID}] ${timestamp}")
    }
}